taf/updater/updater_pipeline.py

```diff
@@ -668,21 +668,59 @@ def validate_out_of_band_and_update_type(self):
             return UpdateStatus.FAILED
 
     def _validate_last_validated_commit(self, last_validated_commit):
-        branch = self.state.users_auth_repo.default_branch
-        users_head_sha = self.state.users_auth_repo.top_commit_of_branch(branch)
-        if last_validated_commit != users_head_sha:
-            # if a user committed something to the repo or manually pulled the changes
-            # last_validated_commit will no longer match the top commit, but the repository
-            # might still be completely valid
-            # committing without pushing is not valid
-            # user_head_sha should be newer than last validated commit
-            commits_since = self.state.users_auth_repo.all_commits_since_commit(
+        branch = self.state.validation_auth_repo.default_branch
+
+        validation_repo = _clone_validation_repo(self.url)
+
+        users_head_sha = validation_repo.top_commit_of_branch(branch)
+
+        def validate_commit_in_remote(repo, commit_sha):
+            try:
+                repo.fetch()
+                remote_commits = repo.all_commits_on_branch(repo.default_branch)
+                return commit_sha in remote_commits
+            except GitError:
+                return False
+
+        # If there's no last validated commit, start validation from the first commit
+        if not last_validated_commit:
+            first_commit_sha = validation_repo.get_first_commit_on_branch(branch)
+            last_validated_commit = first_commit_sha
+
+        # Check if the last validated commit is in the temp repo's history first
+        if last_validated_commit not in validation_repo.all_commits_on_branch(branch):
+            # If not in temp history, check if it's in the remote history
+            if not validate_commit_in_remote(validation_repo, last_validated_commit):
+                msg = f"Last validated commit {last_validated_commit} is not in the remote repository."
+                if self.force:
+                    if validate_commit_in_remote(validation_repo, users_head_sha):
```

Comment:
I think that this should all work without the force flag, since we're not modifying the user's (client's) repo. The temp repo should contain all remote commits, since we clone and fetch from remote.